Code/Framework/AzCore/AzCore/Memory/BestFitExternalMapSchema.cpp

```diff
@@ -9,194 +9,201 @@
-            char* nextEnd = nextStart + nextBlock->first;
-            if (curStart == nextEnd)
+            char* curStart = curBlock->second;
+            char* curEnd = curStart + curBlock->first;
+            bool isMerge = false;
+            for (FreeMapType::iterator nextBlock = curBlock++; nextBlock != m_freeChunksMap.end();)
             {
-                // merge
-                size_t newBlockSize = curBlock->first + nextBlock->first;
-                char* newBlockAddress = nextStart;
-                m_freeChunksMap.erase(nextBlock);
-                FreeMapType::iterator toErase = curBlock;
-                ++curBlock;
-                m_freeChunksMap.erase(toErase);
-                FreeMapType::iterator newBlock = m_freeChunksMap.insert(AZStd::make_pair(newBlockSize, newBlockAddress)).first;
-                if (curBlock != m_freeChunksMap.end() && newBlockSize < curBlock->first)  // if the newBlock in before the next in the list, update next in the list to current
+                char* nextStart = nextBlock->second;
+                char* nextEnd = nextStart + nextBlock->first;
+                if (curStart == nextEnd)
                 {
-                    curBlock = newBlock;
+                    // merge
+                    size_t newBlockSize = curBlock->first + nextBlock->first;
+                    char* newBlockAddress = nextStart;
+                    m_freeChunksMap.erase(nextBlock);
+                    FreeMapType::iterator toErase = curBlock;
+                    ++curBlock;
+                    m_freeChunksMap.erase(toErase);
+                    FreeMapType::iterator newBlock = m_freeChunksMap.insert(AZStd::make_pair(newBlockSize, newBlockAddress)).first;
+                    if (curBlock != m_freeChunksMap.end() &&
+                        newBlockSize <
+                            curBlock->first) // if the newBlock in before the next in the list, update next in the list to current
+                    {
+                        curBlock = newBlock;
+                    }
+                    isMerge = true;
+                    break;
                 }
-                isMerge = true;
-                break;
+                else if (curEnd == nextStart)
+                {
+                    // merge
+                    size_t newBlockSize = curBlock->first + nextBlock->first;
+                    char* newBlockAddress = curStart;
+                    m_freeChunksMap.erase(nextBlock);
+                    FreeMapType::iterator toErase = curBlock;
+                    ++curBlock;
+                    m_freeChunksMap.erase(toErase);
+                    FreeMapType::iterator newBlock = m_freeChunksMap.insert(AZStd::make_pair(newBlockSize, newBlockAddress)).first;
+                    if (curBlock != m_freeChunksMap.end() &&
+                        newBlockSize <
+                            curBlock->first) // if the newBlock in before the next in the list, update next in the list to current
```

Comment:
suggestion: Could we move the comment to above the if and then this line won't be formatted so strangely 😝